### ISSUE TITLE
let declaration not immediately within block

### ISSUE THREAD

**Issue by willlma** [fbzhong/sublime-jslint#35]
I get that error using this code:    

```
function method() {
  'use strict';
  if (true) for (var i = 10; i > 0; i--) {
    let foo = i;
    console.log(foo);
  }
}
```

Let is directly within the for-loop block scope, so should not throw the error (running the above code in Firefox and Chrome works without error).

**Comment by willlma** [comment]
Since this is an error and not a warning, I can't suppress it in `.jshintrc` or code comments.

[Docs](http://jshint.com/docs/):

> `/* jshint -W034 */`
> This syntax works only with warnings (code starts with W), it doesn't work with errors (code starts with E).

**Comment by ghost** [comment]
`.jshintrc` and `/* jshint */` comments are read by [JSHint](https://jshint.com) only, and Sublime-JSLint supports only [JSLint](https://jslint.com) (as the name might suggest). As the JSLint4Java executable does not seem to support ES6 statements (of which `let` is one), I would advise that you install the [jslint module](https://github.com/reid/node-jslint) for [node.js](https://nodejs.org) if you have not already; also be sure to enable the `--edition=latest` flag in Sublime-JSLint's settings (`"node_jslint_options": ""`) so that the most up-to-date version of JSLint is utilised. I believe that JSHint handles ES6 features much better than JSLint, however, so I would probably recommend that you utilise JSHint instead.
